File: src/backend/Heartbeat.Domain/Verification/VerificationStatus.cs
namespace Heartbeat.Domain.Verification;

public class VerificationStatus
{
    public required Guid Id { get; set; } = Guid.NewGuid();

    public required Guid VerificationConfigurationId { get; set; }

    public virtual VerificationConfiguration VerificationConfiguration { get; set; } = null!;

    public required DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public required VerificationToken VerificationToken { get; set; }

    public required VerificationStrategy VerificationStrategy { get; set; }

    public required bool WasVerificationSuccessful { get; set; }

}

File: src/frontend/src/app/pages/apps/models/verification-status.ts
import { VerificationStrategy } from './verification-strategy';

export class VerificationStatus {
  public constructor(
    public verificationStatusId: string,
    public verificationToken: string,
    public verificationStrategy: VerificationStrategy,
    public wasVerificationSuccessful: boolean
  ) {}

  public static from(data: VerificationStatus) {
    console.log('VerificationStatus.from', data);
    return new VerificationStatus(
      data.verificationStatusId,
      data.verificationToken,
      data.verificationStrategy,
      data.wasVerificationSuccessful
    );
  }
}
